src/agents/parking_agent.py

```diff
@@ -25,119 +27,138 @@ def __init__(self, jid, password, x, y, parking_spots, parking_price=40):
-                max_available = min(max_available, self._available_parking_spots[i])
-
-            return max_available
+                logger.info(f"{str(self.agent.jid)}: Reply to {msg.sender} sent: {reply_body.model_dump_json()}")
 
     class MakeReservation(CyclicBehaviour):
         """Behaviour for answering reservation requests"""
 
         async def run(self):
             msg = await self.receive(timeout=PARKING_AGENT_MESSAGE_TIMEOUT)
             if msg:
-                request = RequestReservation(**json.loads(msg.body))
-                reservation_status = self.try_to_reserve_parking_spot(request.time_start, request.time_stop)
-                reservation_id = self.generate_reservation_id() if reservation_status else ""
-
-                reply_body = ReservationResponse(reservation_status, request.user_id, reservation_id)
+                request = RequestReservation.model_validate_json(msg.body)
+                logger.info(f"{str(self.agent.jid)}: Received RequestReservation: {request.model_dump_json()}")
+                reservation_status = self.agent.try_to_reserve_parking_spot(request.time_start, request.time_stop)
+                reservation_id = self.agent.generate_reservation_id(request.user_id) if reservation_status else ""
+                self.agent.store_parking_info_for_user(
+                    request.user_id, reservation_id, request.time_start, request.time_stop)
+
+                reply_body = ReservationResponse(success=reservation_status,
+                                                 user_id=request.user_id, reservation_id=reservation_id)
                 reply = Message(
-                    to=msg.sender,
-                    body=str(reply_body.dict()),
+                    to=str(msg.sender),
+                    body=reply_body.model_dump_json(),
                     thread=msg.thread,
-                    metadata={"performative": "inform", "action": "make-reservation"},
+                    metadata={"performative": "inform", "action": "reservation-response"},
                 )
                 await self.send(reply)
-                print(f"Reply sent: {reply}")
-            else:
-                print("No message received within the timeout period")
-
-        def try_to_resere_parking_spot(self, time_start, time_stop):
-            if self.get_available_parking_spots(time_start, time_stop):
-                for i in range(time_start, time_stop):
-                    self._available_parking_spots[i] -= 1
-                return True
-            return False
-
-        def get_available_parking_spots(self, time_start, time_stop):
-            max_available = self._parking_spots
-            for i in range(time_start, time_stop):
-                max_available = min(max_available, self._available_parking_spots[i])
-
-            return max_available
-
-        def generate_reservation_id(length=12):
-            # @TODO make some kind of hash based on the time and user id
```

Comment:
Maybe let's leave this TODO not to forget about this hash later on

Reply:
i have done generating UUID ;)

Reply:
great, haven't noticed that